docs/how-to-guides/admin/gitlab.rst

```diff
@@ -0,0 +1,107 @@
+
+  * If you *do not* have admin access to GitLab you can:
+     * Use GitLab as IdP for Renku
+     * Have separate Renku and GitLab identities
+   
+  * If you *do* have admin access to GitLab you can:
+     * Use GitLab as IdP for Renku
+     * Use Renku's Keycloak as IdP for GitLab
+     * Have separate Renku and GitLab identities
+
+
+Configuring an External GitLab as Identity Provider for Renku
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the Renku deployment is relying on an existing GitLab instance, this instance could be configured as an identity provider for Renku's Keycloak. This way all of the users from the existing GitLab instance can log into Renku with their existing login information.
+
+#. Go to \https://<renku-domain>/auth (or to your stand-alone Keycloak dashboard), login to ``Admin console`` using ``admin`` as username and the decoded password stored in ``keycloak-password-secret``.
+#. Add an ``Identity Provider`` of type ``OpenID Connect v1.0``.
+#. Set ``Alias`` to <renku-domain>, ``Authorization URL`` to \https://<gitlab-domain>/oauth/authorize, ``Token URL`` to \https://<gitlab-domain>/oauth/token and ``Client ID`` and ``Client Secret`` to the values used in step 1.
+#. Click ``Save``. The new Identity Provider should appear and any user from the stand-alone GitLab instance should be able to login.
+
+To use an external GitLab as a backend for a Renku deployment you must register
+the Renku application in GitLab. If you are not an admin user, this is done under
+your personal user preferences. It is possible that the GitLab instance you are
+using prevents users from registering external applications - in this case you
+must contact your GitLab server administrator for assistance.
+
+
+Register the Renku application in GitLab
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use an external GitLab as a backend for a Renku deployment you must register
+the Renku application in GitLab. If you are not an admin user, this is done under
+your personal user preferences. It is possible that the GitLab instance you are
+using prevents users from registering external applications - in this case you
+must contact your GitLab server administrator for assistance.
+
+
+Renku GitLab application details
+
+Name: renku
+
+Callback URLs:
+
+.. code-block:: console
+
+  https://<your-renku-dns>/login/redirect/gitlab
+  https://<your-renku-dns>/api/auth/gitlab/token
+
+Scopes:
+
+.. code-block:: console
+
+    api (Access the authenticated user's API)
+    read_user (Read the authenticated user's personal information)
+    read_repository (Allows read-access to the repository)
+    read_registry (Grants permission to read container registry images)
+    openid (Authenticate using OpenID Connect)
+
+Copy the secret and client ID and paste it to the dedicated places in renku-values.yaml (step 3)
+
+
+GitLab deployed as part of Renku
+--------------------------------
+
```

Comment:
There is no explanation at all on how to setup Renku as an identity provider to Gitlab. Please add that.

Explain that when you use the Gitlab deployment that comes in Renku this is done for you. But if you use the helm chart then you have to change some things. If you can avoid too many details here and just link to the relevant gitlab documentation page that is the best.

But still I believe that the setup of this is not complicated. But it is definitely worth pointing out.